js/ui/data_grid/ui.data_grid.editing.js

```diff
@@ -1152,6 +1156,10 @@ exports.EditingController = gridCore.ViewController.inherit((function() {
                             if(column.formItem) {
                                 $.extend(item, column.formItem);
                             }
+
+                            if(!that._firstFormItem) {
+                                that._firstFormItem = item;
```

Comment:
The _firstFormItem should not be the invisible item. Also, the _firstFormItem variable should be recalculated after every form rendering.